pkg/scheduler/queue/queue.go

```diff
@@ -245,41 +248,37 @@ func (q *RequestQueue) tryDispatchRequest(broker *queueBroker, call *nextRequest
 	}
 
 	call.lastUserIndex.last = idx
-	if queue == nil {
+	if req == nil {
 		// Nothing available for this querier, try again next time.
 		return false
 	}
 
-	// Pick next request from the queue. The queue is guaranteed not to be empty because we remove empty broker.
-	queueElement := queue.Front()
-
-	requestSent := call.send(nextRequestForQuerier{
-		req:           queueElement.Value,
+	reqForQuerier := nextRequestForQuerier{
+		req:           req,
 		lastUserIndex: call.lastUserIndex,
 		err:           nil,
-	})
+	}
+	requestSent := call.send(reqForQuerier)
 
 	if requestSent {
-		// If GetNextRequestForQuerier received the request, remove it from the queue.
-		// (GetNextRequestForQuerier might have already returned if its context was cancelled.)
-		queue.Remove(queueElement)
-
-		if queue.Len() == 0 {
-			broker.deleteQueue(tenantID)
-		}
-
 		q.queueLength.WithLabelValues(string(tenantID)).Dec()
+	} else {
+		if reqToEnqueue, ok := req.(requestToEnqueue); ok {
+			// should never error; any item previously in the queue already passed validation
```

Comment:
Should we log the error in cases where this does return an error? Otherwise it'll look like requests are just disappearing and this would be very difficult to debug.

Reply:
had to get a logger added to this struct but it's now an error log